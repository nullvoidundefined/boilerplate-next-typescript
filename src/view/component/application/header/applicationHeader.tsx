import styled from "styled-components";
import { Button } from "../../../design-system";
import { COLOR, DIMENSION } from "../../../../constant";
import { ApplicationLogo } from "../logo/applicationLogo";

// TODO add conditional typechecking to negate the need for the default onDropDownToggleButtonClick prop

interface ApplicationHeaderProps {
    isAuth: boolean;
    isMobile: boolean;
    onLogInButtonClick: () => void;
    onLogOutButtonClick: () => void;
    onLogoClick: () => void;

    isNavigationDropdownOpen?: boolean;
    onDropDownToggleButtonClick?: () => void;
}

interface StyledApplicationHeaderProps {
    isMobile: boolean;
}

const StyledButtonContainer = styled.div`
    display: flex;
    gap: 10px;
`;

const StyledApplicationHeader = styled.header<StyledApplicationHeaderProps>`
    align-items: center;
    background-color: ${COLOR.HEADER_BACKGROUND};
    display: flex;
    height: ${({ isMobile }) =>
        isMobile
            ? `${DIMENSION.HEADER_HEIGHT_MOBILE}px`
            : `${DIMENSION.HEADER_HEIGHT_DESKTOP}px`};
    justify-content: space-between;
    padding: 10px;
`;

const ApplicationHeader = ({
    isAuth,
    isMobile,
    isNavigationDropdownOpen,
    onDropDownToggleButtonClick = () => null,
    onLogInButtonClick,
    onLogOutButtonClick,
    onLogoClick,
}: ApplicationHeaderProps) => {
    return (
        <StyledApplicationHeader isMobile={isMobile}>
            <ApplicationLogo onClick={onLogoClick} />
            <StyledButtonContainer>
                {isAuth ? (
                    <Button onClick={onLogOutButtonClick}>
                        <span>Log Out</span>
                    </Button>
                ) : (
                    <Button onClick={onLogInButtonClick}>
                        <span>Log In</span>
                    </Button>
                )}
                {isMobile ? (
                    <Button onClick={onDropDownToggleButtonClick}>
                        {isNavigationDropdownOpen ? "Close Menu " : "Open Menu"}
                    </Button>
                ) : null}
            </StyledButtonContainer>
        </StyledApplicationHeader>
    );
};

export { ApplicationHeader };
