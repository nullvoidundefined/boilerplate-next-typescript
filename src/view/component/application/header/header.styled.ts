import styled from "styled-components";
import { APPLICATION_COLOR, APPLICATION_DIMENSION } from "../../../../constant";

interface StyledHeaderProps {
  isMobile: boolean;
}

const StyledHeader = styled.header<StyledHeaderProps>`
  background-color: ${APPLICATION_COLOR.HEADER_BACKGROUND};
  height: ${APPLICATION_DIMENSION.HEADER_HEIGHT_DESKTOP}px;
  height: ${({ isMobile }) =>
    isMobile
      ? `${APPLICATION_DIMENSION.HEADER_HEIGHT_MOBILE}px`
      : `${APPLICATION_DIMENSION.HEADER_HEIGHT_DESKTOP}px`};
`;

export { StyledHeader };
