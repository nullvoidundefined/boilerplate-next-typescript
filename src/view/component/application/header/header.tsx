import styled from "styled-components";
import { Button } from "../../../design-system";
import { APPLICATION_COLOR, APPLICATION_DIMENSION } from "../../../../constant";

// TODO add conditional typechecking to negate the need for the default onDropDownToggleButtonClick prop

interface HeaderProps {
  isAuth: boolean;
  isMobile: boolean;
  onLogInButtonClick: () => void;
  onLogOutButtonClick: () => void;

  isNavigationDropdownOpen?: boolean;
  onDropDownToggleButtonClick?: () => void;
}

interface StyledHeaderProps {
  isMobile: boolean;
}

const StyledButtonContainer = styled.div`
  display: flex;
  gap: 10px;
`;

const StyledHeader = styled.header<StyledHeaderProps>`
  align-items: center;
  background-color: ${APPLICATION_COLOR.HEADER_BACKGROUND};
  display: flex;
  height: ${({ isMobile }) =>
    isMobile
      ? `${APPLICATION_DIMENSION.HEADER_HEIGHT_MOBILE}px`
      : `${APPLICATION_DIMENSION.HEADER_HEIGHT_DESKTOP}px`};
  justify-content: space-between;
  padding: 10px;
`;

const Header = ({
  isAuth,
  isMobile,
  isNavigationDropdownOpen,
  onDropDownToggleButtonClick = () => null,
  onLogInButtonClick,
  onLogOutButtonClick,
}: HeaderProps) => {
  return (
    <StyledHeader isMobile={isMobile}>
      <h1>VIBE BIO</h1>
      <StyledButtonContainer>
        {isAuth ? (
          <Button onClick={onLogOutButtonClick}>
            <span>Log Out</span>
          </Button>
        ) : (
          <Button onClick={onLogInButtonClick}>
            <span>Log In</span>
          </Button>
        )}
        {isMobile ? (
          <Button onClick={onDropDownToggleButtonClick}>
            {isNavigationDropdownOpen ? "Close Menu " : "Open Menu"}
          </Button>
        ) : null}
      </StyledButtonContainer>
    </StyledHeader>
  );
};

export { Header };
