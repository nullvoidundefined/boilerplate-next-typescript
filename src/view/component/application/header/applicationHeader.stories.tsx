import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { ApplicationHeader } from "./applicationHeader";

export default {
  title: "Component/Application/Header",
  component: ApplicationHeader,
} as ComponentMeta<typeof ApplicationHeader>;

const Template: ComponentStory<typeof ApplicationHeader> = (args) => <ApplicationHeader {...args} />;

export const Desktop = Template.bind({});
Desktop.args = { isMobile: false };

export const Mobile = Template.bind({});
Mobile.args = { isMobile: true };
