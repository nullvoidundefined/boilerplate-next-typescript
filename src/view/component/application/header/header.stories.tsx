import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Header } from "./header";

export default {
  title: "Component/Application/Header",
  component: Header,
} as ComponentMeta<typeof Header>;

const Template: ComponentStory<typeof Header> = (args) => <Header {...args} />;

export const Desktop = Template.bind({});
Desktop.args = { isMobile: false };

export const Mobile = Template.bind({});
Mobile.args = { isMobile: true };
