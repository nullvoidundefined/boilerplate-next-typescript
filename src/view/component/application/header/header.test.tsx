import { Header } from "./header";
import renderer from "react-test-renderer";
import { noop } from "lodash";
import 'jest-styled-components'

it("renders correctly", () => {
  const props = {
    isAuth: true,
    isMobile: true,
    onLogInButtonClick: () => noop(),
    onLogOutButtonClick: () => noop(),
  };
  const component = renderer.create(<Header {...props} />).toJSON();
  expect(component).toMatchSnapshot();
});
