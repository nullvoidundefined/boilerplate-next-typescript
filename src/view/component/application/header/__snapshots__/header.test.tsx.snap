// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 40px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: red;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 60px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 10px;
}

<header
  className="c0"
>
  <h1>
    VIBE BIO
  </h1>
  <div
    className="c1"
  >
    <button
      className="c2 btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span>
        Log Out
      </span>
    </button>
    <button
      className="c2 btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Open Menu
    </button>
  </div>
</header>
`;
